apps/web/src/app/(signing)/sign/[token]/sign-dialog.tsx

```diff
@@ -0,0 +1,82 @@
+import React, { useEffect, useState } from 'react';
+
+import { Loader } from 'lucide-react';
+
+import { Document, Field } from '@documenso/prisma/client';
+import { Button } from '@documenso/ui/primitives/button';
+import {
+  Dialog,
+  DialogContent,
+  DialogFooter,
+  DialogTrigger,
+} from '@documenso/ui/primitives/dialog';
+
+export type SignDialogProps = {
+  isSubmitting: boolean;
+  showConfirmSignatureDialog: boolean;
+  onSignatureComplete: () => void;
+  // eslint-disable-next-line no-unused-vars
+  setShowConfirmSignatureDialog: (show: boolean) => void;
+  document: Document;
+  fields: Field[];
+};
+
+export default function SignDialog({
+  isSubmitting,
+  showConfirmSignatureDialog,
+  onSignatureComplete,
+  setShowConfirmSignatureDialog,
+  document,
+  fields,
+}: SignDialogProps) {
+  const [isComplete, setIsComplete] = useState(false);
+
+  useEffect(() => {
+    const isComplete = fields.every((f) => f.inserted);
+    setIsComplete(isComplete);
+  }, [fields]);
+
+  return (
+    <Dialog open={showConfirmSignatureDialog} onOpenChange={setShowConfirmSignatureDialog}>
+      <DialogTrigger asChild>
+        <Button className="w-full" type="button" size="lg" disabled={!isComplete || isSubmitting}>
```

Comment:
You can use the `loading` prop to handle displaying the spinner on `isSubmitting`